### PR TITLE
FLA-764 re-implement Package Confirmation file list

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

https://justice.gov.bc.ca/jira/browse/FLA-764
- Re-implemented the file list on the Package Confirmation screen
- Refactored to smaller sub-components

Eventually, its would be nice to have the Package Review screen use the same sub-components. We'll do this on a separate PR to keep them smaller.

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

yarn lint
yarn test

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by alexjoybc** [comment]
Agree with you on the component for the list @KDNeufeld

File: src/frontend/efiling-frontend/src/domain/package/package-confirmation/PackageConfirmation.js [hunk 1]
```diff
@@ -198,28 +129,9 @@ export default function PackageConfirmation({
...
-                  }}
-                  elements={generateTableData(file, submissionId)}
-                />
-              }
-            />
-            <br />
-          </div>
-        ))}
-        <h2>
+        {<FileList submissionId={submissionId} files={files} />}
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/domain/package/package-confirmation/PackageConfirmation.js:132]: Nice!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

File: src/frontend/efiling-frontend/src/domain/package/package-confirmation/PackageConfirmationService.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import axios from "axios";
+import FileSaver from "file-saver";
+
+/**
+ * Retrieves the Uploaded Document from the backend API for the given submissionId and filename.
+ *
+ * @param submissionId the submissionId associated with the package
+ * @param file the document to retrieve. Shape must be { mimeType: "string", documentProperties: { name: "string" } }
+ */
+export const downloadFile = async (submissionId, file) => {
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/domain/package/package-confirmation/PackageConfirmationService.js:10]: Should this go in the document service ?

File: src/frontend/efiling-frontend/src/modules/helpers/CurrencyUtil.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Dinero from "dinero.js";
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/modules/helpers/CurrencyUtil.js:1]: never heard of this one before ;-)

**Review by alexjoybc — CHANGES_REQUESTED** [review]
Awesome work @KDNeufeld , the FileList looks really good, I just have one comment about the download function if that should not be part of the document service.. let me know what you think.

File: src/frontend/efiling-frontend/src/modules/helpers/CurrencyUtil.js (hunk 3 above)
Comment (KDNeufeld) [reply src/frontend/efiling-frontend/src/modules/helpers/CurrencyUtil.js:1]: Nope, me neither. That's just what the app was using everywhere, so I refactored it out to a common util method so we can test it

File: src/frontend/efiling-frontend/src/domain/package/package-confirmation/PackageConfirmationService.js (hunk 2 above)
Comment (KDNeufeld) [reply src/frontend/efiling-frontend/src/domain/package/package-confirmation/PackageConfirmationService.js:10]: Yes, that's a good thought. I'll get on this.

**Review by alexjoybc — APPROVED** [review]
Awesome!!! thanks @KDNeufeld

**Pull request merged**